dashboard/src/shared/AppRepository.ts

```diff
@@ -116,17 +117,20 @@ export class AppRepository {
     ociRepositories: string[],
     skipTLS: boolean,
   ) {
-    const { data } = await axiosWithAuth.post<any>(url.backend.apprepositories.validate(cluster), {
-      appRepository: {
-        repoURL,
-        type,
-        authHeader,
-        authRegCreds,
-        customCA,
-        ociRepositories,
-        tlsInsecureSkipVerify: skipTLS,
+    const { data } = await axiosWithAuth.post<any>(
+      url.backend.apprepositories.validate(cluster, namespace),
+      {
+        appRepository: {
+          repoURL,
+          type,
+          authHeader,
```

Comment:
I'm still catching up after being away, but isn't the credential secret already created on the cluster by the time the user submits/validates the repo? If the UX doesn't create this particular secret until after the repo is created, then OK, but otherwise, I'm not sure why we send it over the wire rather than referring to the secret? This leaks the credential to the browser when it potentially never needs to leave the cluster (eg. when using a pre-existing secret).

Reply:
yes, `authRegCreds` is the name of the secret not the secret content (like it's the case for `authHeader`)

Reply:
But `authHeader` is an actual credential (that will be saved as a secret after validation on submission?)